File: C - 2D Arrays/Solution.cs
using System.CodeDom.Compiler;
using System.Collections.Generic;
using System.Collections;
using System.ComponentModel;
using System.Diagnostics.CodeAnalysis;
using System.Globalization;
using System.IO;
using System.Linq;
using System.Reflection;
using System.Runtime.Serialization;
using System.Text.RegularExpressions;
using System.Text;
using System;



class Solution
{
    public static void Main(string[] args)
    {
        
        int[] sum = new int[16];
        int hourglass = 0;
        
        List<List<int>> arr = new List<List<int>>();
        
       

        for (int i = 0; i < 6; i++)
        {
            arr.Add(Console.ReadLine().TrimEnd().Split(' ').ToList().Select(arrTemp => Convert.ToInt32(arrTemp)).ToList());
           
        }
        
        for (int i = 0; i < 4; i++)
        {
            for (int j = 0; j < 4; j++)
            {
                sum[hourglass] = arr[i][j] + arr[i][j + 1] + arr[i][j + 2] 
                + arr[i + 1][j + 1] + arr[i + 2][j] 
                + arr[i + 2][j + 1] + arr[i + 2][j + 2];
                hourglass++;
            }
        }
        
        Array.Sort(sum);
        Console.WriteLine(sum[15]);
        
    }
        
}

File: C - 2D Arrays/Solution.fs
open System

[<EntryPoint>]
let main argv = 
    let arr = 
        [| [| 0; 0; 0; 0; 0; 0 |]
           [| 0; 0; 0; 0; 0; 0 |]
           [| 0; 0; 0; 0; 0; 0 |]
           [| 0; 0; 0; 0; 0; 0 |]
           [| 0; 0; 0; 0; 0; 0 |]
           [| 0; 0; 0; 0; 0; 0 |] |]
    for i = 0 to 5 do
        arr.[i] <- Console.ReadLine().Split(' ') |> Array.map int
    let mutable max = -9 * 7
    for i = 0 to 5 do
        for j = 0 to 5 do
            if j + 2 < 6 && i + 2 < 6 then 
                let sum = 
                    arr.[i].[j] + arr.[i].[j + 1] + arr.[i].[j + 2] + arr.[i + 1].[j + 1] + arr.[i + 2].[j] 
                    + arr.[i + 2].[j + 1] + arr.[i + 2].[j + 2]
                if sum > max then max <- sum
    Console.WriteLine(max)
    0
